connectors/src/connectors/notion/lib/notion_api.ts

```diff
@@ -181,6 +181,48 @@ export async function getPagesEditedSince(
   };
 }
 
+export async function isPageAccessibleAndUnarchived(
+  notionAccessToken: string,
+  pageId: string,
+  localLogger?: Logger
+) {
+  const notionClient = new Client({ auth: notionAccessToken });
+  const maxTries = 5;
+  let tries = 0;
+
+  while (tries < maxTries) {
+    const tryLogger = (localLogger || logger).child({
+      tries,
+      maxTries,
+      pageId,
+    });
+    try {
+      tryLogger.info("Checking if page is accessible and unarchived.");
+      const page = await notionClient.pages.retrieve({ page_id: pageId });
+      if (!isFullPage(page)) {
+        return false;
+      }
+      return !page.archived;
+    } catch (e) {
+      if (APIResponseError.isAPIResponseError(e)) {
+        if (["rate_limited", "internal_server_error"].includes(e.code)) {
```

Comment:
As per IRL discussion, let's white list errors not black list (in case Notion goes wackooo)

Reply:
Done